src/Microsoft.NetCore.Analyzers/UnitTests/Runtime/DoNotLockOnObjectsWithWeakIdentityTests.cs

```diff
@@ -264,6 +271,126 @@ End Sub
             GetCA2002BasicResultAt(18, 30, "System.ExecutionEngineException"));
         }
 
+        [Fact]
+        [WorkItem(2744, "https://github.com/dotnet/roslyn-analyzers/issues/2744")]
+        public async Task CA2002_MonitorEnter_Diagnostic()
+        {
+            await VerifyCS.VerifyAnalyzerAsync(@"
+using System.Threading;
+
+public class C
+{
+    public void Foo()
+    {
+        Monitor.Enter(this);
```

Comment:
I have only tested for `this` and `string` but if you prefer I could add a test for all supported types.

Reply:
I think it is fine.